libs/blockchain/src/blockchain-rpc-query.service.ts

```diff
@@ -237,6 +245,23 @@ export class BlockchainRpcQueryService extends PolkadotApiService {
     };
   }
 
+  public async checkTxCapacityLimit(providerId: AnyNumber, encodedExt: HexString): Promise<boolean> {
+    const capacityInfo = await this.capacityInfo(providerId);
+    const { remainingCapacity } = capacityInfo;
+
+    // Calculate the total capacity cost for all encoded extensions
+    let totalAdjustedWeightFee = BigInt(0);
+    const capacityCost = await this.getCapacityCostForExt(encodedExt);
+    if (capacityCost.inclusionFee.isSome) {
+      const inclusionFee = capacityCost.inclusionFee.unwrap();
+      totalAdjustedWeightFee += inclusionFee.adjustedWeightFee.toBigInt();
+    }
+
+    // Check if the total cost exceeds the remaining capacity
+    const outOfCapacity = remainingCapacity < totalAdjustedWeightFee;
+    return outOfCapacity;
+  }
+
```

Comment:
this calls `rpc` to get `FeeDetails` which has adjusted fee that should be less than remaining capacity. this is not a breaking change if rpc fails for some bad transaction and returns null this logic will default to false